### PR TITLE
Safari implementation

### PR DESCRIPTION
**Pull request by imurchie**
There is cleanup to be done, but this is Safari for simulators.

Two parts don't work:
1 asynchronous execute of JS in a frame
2. performance logging

I will look into these under a separate heading.

### CODE REVIEW

**Comment by imurchie** [comment]
Fixed (1) above.

**Comment by sebv** [comment]
What's the commented merge part for?

**Comment by imurchie** [comment]
It was commented out, rather than deleted, while merging, since I didn't know where you had changed things. I need to delete it now.

File: lib/commands/web.js [hunk 1]
```diff
@@ -53,39 +196,201 @@ commands.findWebElementOrElements = async function (strategy, selector, ctx, man
...
+  // add static offset for safari in landscape mode
+  let yOffset = this.curOrientation === 'LANDSCAPE' ? this.landscapeWebCoordsOffset : 0;
+
+  // absolutize web coords
+  let webviews = await this.uiAutoClient.sendCommand(wvCmd);
+  if (webviews.length < 1) {
+    throw new errors.UnknownError.code('Could not find any webviews to click inside!');
+  }
+  if (_.isUndefined(webviews[webviewIndex])) {
+    logger.warn("Could not find webview at index " + webviewIndex + ", " +
```
Comment (scottdixon) [inline lib/commands/web.js]: ${webviewIndex}

File: lib/commands/gesture.js [hunk 2]
```diff
@@ -27,7 +28,10 @@ commands.click = async function (el) {
 
 commands.clickCurrent = async function (/*button*/) {
   if (this.isWebContext()) {
-    /* TODO */throw new errors.NotYetImplementedError();
+    if (_.isNull(this.curWebCoords)) {
```
Comment (jlipps) [inline lib/commands/gesture.js]: random comment. I think I prefer `this.curWebCoors === null`, unless `_.isNull` does something I'm not aware of

**Pull request merged**